java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -365,74 +379,47 @@ abstract class AbstractArcHost(
     }
 
     /**
-     * Invokes necessary [Particle] lifecycle methods given the current
-     * [ParticleContext.particleState], and changes that state if necessary. For example by
-     * insuring that [Particle.onFirstStart()], [Particle.onShutdown()] are properly called.
+     * Invokes necessary [Particle] startup lifecycle methods given the current
+     * [ParticleContext.particleState], and changes that state as necessary.
      */
     private suspend fun performParticleLifecycle(particleContext: ParticleContext) {
         if (particleContext.particleState == ParticleState.Instantiated) {
             try {
-                // onFirstStart() must succeed, else we consider the particle startup failed
                 particleContext.particle.onFirstStart()
```

Comment:
both here, and down below when onStart() is invoked, I think these should be scope.launch(dispatcher) { ... }

You can get a dispatcher via ScheduleDispatcher(schedulerProvider(arcId))

However that makes it possible for performParticleLifecycle() to return before onFirstStart/onStart() is invoked,  so we will need to .join() I think. Jason, am I correct?

Reply:
Thanks Ray, I wasn't sure about the dispatching side of things. Does `ParticleContext.notify` also need to use a dispatcher? That only gets invoked by the StorageProxy's scheduler, so I has assumed that was ok.

Could we potentially have a bad sequencing interaction if `onFirstStart` and `onStart` are launched with a different scheduler than the rest of the lifecycle methods?

Reply:
Done.